### DISCUSSION TITLE
Multipart exrs crashing Nuke

### DISCUSSION THREAD

**Discussion by noidea-man** [OpenImageIO/oiio#3093] [Q&A] [answered]
Hi!
I'm running into an issue with multipart exrs that I generate through oiiotool.  I have a workaround of sorts, so this isn't super critical, but I'm extremely curious if I'm doing something wrong or if anyone else has run into this.  FWIW, I'm on windows currently,  using this to convert multichannel renders from Arnold, testing in Nuke 11, 12, and 13.  

I've been building my workflow against the oiiotool.exe that is packed with the version of Arnold I've been using.  It appears to be oiiotool version 2.1.0 .  Now that I have that working, I wanted to move all my tools up to the latest version, and what I found is that if I try to use any version of oiiotool 2.1.4 or above, the resulting exr will crash Nuke.  The image metadata looks correct, the main rgba channels are correct, but trying to view or manipulate any subimage will crash almost instantly.

Obviously, my workaround is to just stick with 2.1.0 and call it a day, but I'm curious if something changed between those versions that makes sense of all of this. For the record, my arguments look like this:
`.\oiiotool.exe input.exr --ch R=R,B=B,G=G,A=A -attrib oiio:subimagename rgba --ch B=aov1.B,R=aov1.R,G=aov1.G -attrib oiio:subimagename aov1 --eraseattrib ".*" input.exr --ch B=aov2.B,R=aov2.R,G=aov2.G -attrib oiio:subimagename aov2 --eraseattrib ".*" --siappendall -o output.exr`

If I'm reading the changelogs correctly, it uses a newer version of openexr, which I guess I can see causing a problem but I feel like I would not be the first to find that.  Is there a different way to structure my arguments that may prevent this or could it be a result of not compiling oiio myself?

**Comment by lgritz** [comment, marked as answer]
Can you make one of those "broken" exrs and put it someplace where we can download it? I want to see what exactly is wrong with it.

I'm a little concerned about that command line that adds subimage name attribute and then immediately erases all attributes. What is that supposed to do?

**Reply by noidea-man** [reply]
That.....That was the clue I needed and I can't believe I didn't catch that.  The idea was that every subimage needed it's metadata removed, otherwise the full image is even slower in nuke than the original non multipart image.  Reversing the order of those commands does indeed fix my problem.  I'm astonished I got so far down this path using a version that made good images when it never should have!

**Reply by lgritz** [reply]
I still feel like something weird is going on if the resulting file "crashes Nuke." Do you mean it just does the wrong thing, or do you literally mean that Nuke itself crashes and exits immediately upon reading that file?

Can you send me one of the broken files?  So what if you erase all of the attributes before you combine them into one multi-part image? I'm not sure why that would make an invalid file, and whatever's going wrong, we should fix it.

**Reply by noidea-man** [reply]
I'll try to get a render together that doesn't break an NDA asap!  
When I bring one of the broken ones into Nuke, it shows the main rgba channels just fine, and then If I attempt to view one of the subimages it appears black with an "unexpected data block y coordinate" error at the top of screen.  After that it sort of locks up and if I try to interact with another node or the timeline, Nuke exits immediately.

edit:  I was wrong in the first message. Looking a little closer at my broken one, the arguments are getting shuffled up a little more than I thought they were.    I actually had `--eraseattrib ".*" --ch B=aov1.B,R=aov1.R,G=aov1.G -attrib oiio:subimagename aov1 ` , so I believe I was correctly removing the metadata before adding the subimagename, but I was doing it before defining the channels.  Moving the eraseattrib to just before the subimagename does the trick.  Apologies for screwing up the first post, but same end result!

**Reply by lgritz** [reply]
That doesn't make any sense to me, --eraseattrib should have no effect on the channel order or naming and vice versa. I feel like there is something going on that we still haven't quite fully understood. 

I'd love to get a set of (a) input images, (b) oiiotool command line, and (c) the output YOU get with that command (just in case the version I use behaves differently), that result in an output that crashes Nuke or is in an other way wrong. Seems like we should investigate until we can really articulate what the sequence of events is that cause the problem, and why.

**Reply by noidea-man** [reply]
Yeah, agreed.  I definitely want to run through my whole process again from scratch to make sure I have the facts straight.  What's the best way to send all the pieces to you?

**Reply by lgritz** [reply]
Maybe put them on a google or dropbox drive and send me the link via email?

**Reply by lgritz** [reply]
I won't show them to anybody else and will delete them as soon as I'm done. I just want a repro case to investigate this fully.
